netinterface.py

```diff
@@ -175,17 +180,38 @@ def writeRHStyleInterface(self, iface):
             f.write("BOOTPROTO=dhcp\n")
             f.write("PERSISTENT_DHCLIENT=1\n")
         else:
+            f.write("BOOTPROTO=none\n")
+
+        if self.mode == self.Static:
             # CA-11825: broadcast needs to be determined for non-standard networks
             bcast = self.getBroadcast()
-            f.write("BOOTPROTO=none\n")
             f.write("IPADDR=%s\n" % self.ipaddr)
             if bcast is not None:
                 f.write("BROADCAST=%s\n" % bcast)
             f.write("NETMASK=%s\n" % self.netmask)
             if self.gateway:
                 f.write("GATEWAY=%s\n" % self.gateway)
+
+        if self.modev6:
+            with open('/etc/sysconfig/network', 'w') as net_conf:
+                net_conf.write("NETWORKING_IPV6=yes\n")
+            f.write("IPV6INIT=yes\n")
+            f.write("IPV6_DEFROUTE=yes\n")
+            f.write("IPV6_DEFAULTDEV=%s\n" % iface_vlan)
+            f.write("IPV6_AUTOCONF=yes\n" if self.modev6 == self.Autoconf else "IPV6_AUTOCONF=no\n")
+
+        if self.modev6 == self.DHCP:
+            f.write("DHCPV6C=yes\n")
+            f.write("PERSISTENT_DHCLIENT_IPV6=yes\n")
+            f.write("IPV6_FORCE_ACCEPT_RA=yes\n")
+        elif self.modev6 == self.Static:
+            f.write("IPV6ADDR=%s\n" % self.ipv6addr)
+            if self.ipv6_gateway:
+                f.write("IPV6_DEFAULTGW=%s\n" % (self.ipv6_gateway))
```

Comment:
It seems awkward to have the `Autoconf` case handled that way.  It would be more visible that all cases are covered bo have it be a 3rd branch of that `if`, even if that means duplicating the `AUTOCONF=no` line for the other two cases. And we could then add `elif == None: pass else: assert()` to make the code future-robust.